### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by AlexeyKriva**

### CODE REVIEW

File: src/main/java/com/software/modsen/ridesmicroservice/entities/ride/RideDto.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    @PastOrPresent(message = "Date cannot be in the future.")
+    @NotNull(message = "Order date time cannot be null.")
+    @JsonProperty("order_date_time")
+    private LocalDateTime orderDateTime;
+    @NotNull(message = "Price cannot be null.")
+    @JsonProperty("price")
+    private Float price;
+    @NotNull(message = "Currency cannot be null.")
+    @JsonProperty("currency")
+    private Currency currency;
```
Comment (psidarovich) [inline src/main/java/com/software/modsen/ridesmicroservice/entities/ride/RideDto.java:38]: minor: please, use empty lines between params with lots of annotations. Hard to read

File: src/main/java/com/software/modsen/ridesmicroservice/entities/ride/RidePutDto.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    @JsonProperty("ride_status")
+    private RideStatus rideStatus;
+    @PastOrPresent(message = "Date cannot be in the future.")
+    @NotNull(message = "Order date time cannot be null.")
+    @JsonProperty("order_date_time")
+    private LocalDateTime orderDateTime;
+    @NotNull(message = "Price cannot be null.")
+    @JsonProperty("price")
+    private Float price;
+    @NotNull(message = "Currency cannot be null.")
```
Comment (psidarovich) [inline src/main/java/com/software/modsen/ridesmicroservice/entities/ride/RidePutDto.java:37]: Minor: you can omit such messages, spring validation will make them for you

File: src/main/java/com/software/modsen/ridesmicroservice/services/RideService.java [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+public class RideService {
+    @Autowired
+    private RideRepository rideRepository;
+    @Autowired
+    private PassengerClient passengerClient;
```
Comment (psidarovich) [inline src/main/java/com/software/modsen/ridesmicroservice/services/RideService.java:26]: Bad practice to use field injection. How will you test this class (unit tests)?

File: src/main/java/com/software/modsen/ridesmicroservice/services/RideService.java [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+
+    public List<Ride> getAllRides() {
+        return rideRepository.findAll();
+    }
+
+    public List<Ride> getAllNotCompletedAndNotCancelledRides() {
+        return rideRepository.findAll().stream()
+                .filter(ride -> ride.getRideStatus() != RideStatus.COMPLETED
+                        && ride.getRideStatus() != RideStatus.CANCELLED
+                )
```
Comment (psidarovich) [inline src/main/java/com/software/modsen/ridesmicroservice/services/RideService.java:39]: Please, use filtration on db layer, not in memory

**Review by psidarovich — CHANGES_REQUESTED** [review]

File: src/main/java/com/software/modsen/ridesmicroservice/services/RideService.java (hunk 4 above)
Comment (AlexeyKriva) [reply src/main/java/com/software/modsen/ridesmicroservice/services/RideService.java:39]: I updated this query to repository query with JPQL

File: src/main/java/com/software/modsen/ridesmicroservice/services/RideService.java (hunk 3 above)
Comment (AlexeyKriva) [reply src/main/java/com/software/modsen/ridesmicroservice/services/RideService.java:26]: I will fix these rows when i refactor files in my projects

**Review by psidarovich — APPROVED** [review]

**Pull request merged**